library/CM/Http/Request/Abstract.php

```
@@ -278,6 +278,17 @@ public function getQuery() {
     }
 
     /**
+     * @return array
+     */
+    public function findQuery() {
+        try {
+            return $this->getQuery();
+        } catch (CM_Exception_Invalid $e) {
```

>Maybe rather encode the query in the constructor (or better outside) and pass along correct query. We would throw exception there.

hmm.  do you mean _decode_ query? In `Request` constructor? 
Probably it would be good. But it's again about changing of class behaviour. It will fail faster and i guess in most cases it's okay, but in general it could be differently.

However I think we should not inject already parsed query from outside as it is the area of class responsibility along with cookies, headers etc